castep_outputs/parsers/cell_param_file_parser.py

```diff
@@ -387,6 +444,64 @@ def _parse_symops(block: Block) -> list[dict[str, ThreeByThreeMatrix | ThreeVect
              "t": tmp[i+3]}
             for i in range(0, len(tmp), 4)]
 
+def _parse_xc_definition(block: Block) -> dict[str, float]:
+    """
+    Parse xc_definition block to dict.
+
+    Parameters
+    ----------
+    block
+        Input block to parse.
+
+    Returns
+    -------
+    dict[str, float]
+        Parsed info.
+    """
+    block_data = {"xc": {}, "params": {}}
+    block.remove_bounds(fore=0, back=1)
+
+    for line in strip_comments(block):
+        key, val = line.split(maxsplit=1)
+        key = normalise_key(key)
+
+        if key in {"nlxc_screening_length", "nlxc_screening_function",
+                   "nlxc_ppd_int", "nlxc_divergence_corr"}:
+
+            if key == "nlxc_screening_length":
+                block_data["params"][key] = float(val)
+            if key == "nlxc_screening_function":
+                block_data["params"][key] = val
+            if key == "nlxc_ppd_int":
+                block_data["params"][key] = val.upper() == "ON"
+            if key == "nlxc_divergence_corr":
+                block_data["params"][key] = val.upper() == "ON"
+
+            continue
+
+        block_data["xc"][key] = float(val)
```

Comment:
I'm not sure this nested structure with a `continue` is more readable than

```suggestion
        if key == "nlxc_screening_length":
            block_data["params"][key] = float(val)
        elif key == "nlxc_screening_function":
            block_data["params"][key] = val
        elif key in ("nlxc_ppd_int", "nlxc_divergence_corr"):
            block_data["params"][key] = val.upper() == "ON"
        else:
            block_data["xc"][key] = float(val)

```

Reply:
Indeed, part of overly frequent restructuring. Thanks for catching this.